app/src/main/java/org/wikipedia/gallery/GalleryActivity.java

```diff
@@ -323,6 +330,11 @@ public void onActivityResult(int requestCode, int resultCode, final Intent data)
             layOutGalleryDescription();
             setResult(ACTIVITY_RESULT_IMAGE_CAPTION_ADDED);
         }
+        if (requestCode == ACTIVITY_REQUEST_ADD_IMAGE_TAGS && resultCode == RESULT_OK) {
+            FeedbackUtil.showMessage(this, getString(R.string.description_edit_success_saved_image_tags_snackbar));
+            layOutGalleryDescription();
+            setResult(ACTIVITY_REQUEST_ADD_IMAGE_TAGS);
```

Comment:
Not sure if it is really necessary.

Reply:
I am thinking this is linked to the issue you are seeing... this result should be used to refresh. Working on it..